### DISCUSSION TITLE
Auto-respond on specific label

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#25062] [Actions]
<p>I was wondering if there exists a GitHub Action for sending specific canned responses (either from the workflow file itself or perhaps pre-made markdown files) if an issue receives a specific label (Maybe even with actions like closing the issue/PR).</p>
<p>Would it even be beneficial at all to use a GitHub Action for this in the first place, since it requires to have a server/system to be set up and load all materials and resources, etc.</p>
<p>Right now I use a GitHub App (Bot) for doing the auto-message stuff, but I would prefer to switch to GitHub Actions for that if it is possible.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Firstly you need to trigger the workflow with ‘<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#issues-event-issues" rel="nofollow noopener">issue</a>’ event, get the new label name from context value ‘${{ github.event.label.name }}’, according to the label name, determin the respond then.</p>
<p>For instance, below code will check if new label ‘bug’ is added and respond accordingly. Hope it helps!</p>
<pre><code>name: labelrespond
on:
    issues:
        types: labeled

jobs:
  issuelabel:
    runs-on: [ubuntu-latest]
    steps:
      - name: get labels name
        if: contains(github.event.label.name, 'bug' )
        run: |
          echo ${{ github.event.label.name }}
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181088625-e2dee8eb-e0e5-4d34-b539-b4c95fc2d769.png" alt="Labelrespond.png" data-base62-sha1="7ZXh0mRd8OqXFSga0J9njgzervu" width="293" height="424"></p>

**Comment by Andre601** [comment]
<p>Thanks for the answer,</p>
<p>Now I still wonder if it is worth it, to have such a GitHub Action in the first place, due to it requiring a OS to be setup for just a message to be send.</p>
